core/api/views/business_plan.py

```diff
@@ -101,13 +103,30 @@ def create(self, request, *args, **kwargs):
         self.perform_create(serializer)
```

Comment:
I think you should check if the agency user is creating a bp for the correct agency.